### PR TITLE
Fixes #622 - Include version in Jar filenames of Accumulo deps

### LINKED ISSUE #622: Include version in Jar filenames of Accumulo depedencies
At some point, Accumulo moved away from including version numbers in file names for dependencies found in the `lib/` directory of the Accumulo binary release tarball. While I am sure there was a good reason for this at the time, it would be nice to move back to putting the version in the filename of these jars.

### PR DESCRIPTION
**Pull request by mikewalch**
I looked into the original reasoning for removing versions and found this reasoning in [ACCUMULO-2863](https://issues.apache.org/jira/browse/ACCUMULO-2863):

> We strip versions off of dependencies we package in the lib directory of our tarball. This is done for a few beneficial reasons (primarily: avoid classpath polution with multiple versions of the same jar, and ease maintenance of scripts that configure the classpath).

I don't think this reasoning still applies.  However, I am open to closing this PR if others still think omitting version is still necessary.

### CODE REVIEW

**Comment by mikewalch** [comment]
Also, I don't think the `VERSIONS` file is necessary anymore as you can now just get the version from the jar and this file wasn't kept up to date anyways as the dependencies for this file needed to be updated in the assemble/pom.xml

**Comment by mikewalch** [comment]
@ctubbsii, sounds good to me.

File: assemble/src/main/assemblies/component.xml [hunk 1]
```diff
@@ -23,7 +23,7 @@
       <directoryMode>0755</directoryMode>
       <fileMode>0644</fileMode>
       <useProjectArtifact>false</useProjectArtifact>
-      <outputFileNameMapping>${artifact.artifactId}${dashClassifier?}.${artifact.extension}</outputFileNameMapping>
+      <outputFileNameMapping>${artifact.artifactId}-${artifact.version}${dashClassifier?}.${artifact.extension}</outputFileNameMapping>
```
Comment (ctubbsii) [inline assemble/src/main/assemblies/component.xml]: This line should just be deleted. This is the default `outputFileNameMapping`.

Comment (mikewalch) [reply assemble/src/main/assemblies/component.xml]: Fixed in 0247ac376dcf123

**Comment by ctubbsii** [comment]
Yeah, deleting that line was my only feedback.

**Pull request merged**